frameworks/C++/cppcms/src/test_db.cpp

```diff
@@ -0,0 +1,172 @@
+    int count = cinvert_constraint_count(val);
+    cppcms::json::array my_object;
+
+    for (int i = 0; i < count; ++i) {
+        cppcms::json::value my_val;
+        int r = get_random_id();
+
+        world_object wo;
+        if (get_world_object_db(wo, r)) {
+            my_val["id"] = wo.id;
+            my_val["randomNumber"] = wo.randomNumber;
+            int rv = get_random_value();
+            wo.randomNumber = rv;
+            if (!update_world_object_db(wo)) {
+                return;
+            }
+
+            my_object.push_back(my_val);
+        } else {
+            return;
+        }
+    }
+
+    send_json(my_object);
+}
+
+void test_db::servre_cached_worlds(std::string val) {
+    //response().make_error_response(response().not_implemented);
+    int count = cinvert_constraint_count(val);
+    cppcms::json::array my_object;
+    for (int i = 0; i < count; ++i) {
+        cppcms::json::value my_val;
+        int r = get_random_id();
+        world_object wo;
+        if (!cache().fetch_data(std::to_string(r), wo)) {
+            if (get_world_object_db_ch(wo, r)) {
+                cache().store_data(std::to_string(r), wo);
+            } else {
+                return;
+            }
+        }
+        my_val["id"] = wo.id;
+        my_val["randomNumber"] = wo.randomNumber;
+        my_object.push_back(my_val);
+    }
+
+    send_json(my_object);
+
+
+}
+
+int test_db::get_random_id() {
+    return rand() % (MAX_WORLD_ID - 1) + 1;
+}
+
+int test_db::get_random_value() {
+    return rand() % (MAX_RANDOM_VALUE - 1) + 1;
+}
+
+int test_db::cinvert_constraint_count(const std::string & val) {
+    int count = 0;
+    count = atoi(val.c_str());
+    if (count < 1)count = 1;
+    if (count > MAX_QUERY_COUNT)count = MAX_QUERY_COUNT;
+    return count;
+}
+
+bool test_db::get_world_object_db_ch(world_object &c, int id) {
+    //c.id = id;
+    cppdb::result res;
+    res = sql << "SELECT id, randomNumber FROM world WHERE id = ?" << id << cppdb::row;
+    if (res.empty()) {
+        response().make_error_response(response().internal_server_error, "SQL result empty");
+        return false;
+    }
+    res >> c.id >> c.randomNumber;
+    return true;
+}
+
+bool test_db::get_world_object_db(world_object &c, int id) {
+    //c.id = id;
+    cppdb::result res;
+    res = sql << "SELECT id, randomNumber FROM world WHERE id = ?" << id << cppdb::row;
+    if (res.empty()) {
+        response().make_error_response(response().internal_server_error, "SQL result empty");
+        return false;
+    }
+    res >> c.id >> c.randomNumber;
+    return true;
+}
+
+bool test_db::update_world_object_db(world_object &c) {
+    cppdb::statement st = sql
+            << "UPDATE world SET randomNumber = ? WHERE id = ?"
+            << c.randomNumber << c.id;
+    st.exec();
+    //    if (st.affected() == 0) {
+    //        response().make_error_response(response().internal_server_error, "SQL bad update");
+    //        return false;
+    //    }
```

Comment:
There are also a couple of places with commented out code - let's delete these as well.